src/lib/optimizer/join_ordering/abstract_join_ordering_algorithm.hpp

```diff
@@ -12,19 +12,64 @@ class AbstractLQPNode;
 class JoinGraph;
 class CardinalityEstimationCache;
 
+/**
+ * Given a JoinGraph (created by the JoinGraphBuilder), the responsibility of a join ordering algorithm is to bring the
+ * joins into a close-to-optimal order. This means that the total cost of all joins, as estimated by the
+ * AbstractCostEstimator should be minimal.
+ *
+ * JoinOrderingAlgorithms operate on vertices, which are considered to be atomic, as described in the JoinGraph class.
+ * These may be StoredTableNodes, but could also be more complex subplans that are treated as a single table for the
+ * purpose of join ordering.
+ */
 class AbstractJoinOrderingAlgorithm {
  public:
   virtual ~AbstractJoinOrderingAlgorithm() = default;
 
- protected:
-  static std::shared_ptr<AbstractLQPNode> _add_predicates_to_plan(
-      const std::shared_ptr<AbstractLQPNode>& lqp, const std::vector<std::shared_ptr<AbstractExpression>>& predicates,
-      const std::shared_ptr<AbstractCostEstimator>& cost_estimator);
+  /*
+   * @param join_graph      A JoinGraph for a part of an LQP with further subplans as vertices. GreedyOperatorOrdering
+   *                        is only applied to this particular JoinGraph and doesn't modify the subplans in the
+   *                        vertices.
+   * @return                An LQP consisting of
+   *                         * the operations from the JoinGraph in a greedy order, trying to minimize intermediate
+   *                           cardinalities
+   *                         * the subplans from the vertices below them
```

Comment:
Isn't this description too specific for the abstract class?
Either add the DPccp description as well or generalize it a bit.

Reply:
Ah, I missed the greedy order. Will change.